pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -138,6 +145,117 @@ func newManagerImpl(socketPath string) (*ManagerImpl, error) {
 	return manager, nil
 }
 
+//GetTopologyHints calculates a BitMask based on device socket availability for an incoming container
+func (m *ManagerImpl) GetTopologyHints(pod v1.Pod, container v1.Container) topologymanager.TopologyHints {
```

Comment:
It might be good to move this function to its own file and split it in multiple sub functions to help with readability

Reply:
Sure, I'll do that!